test/src/unit-dense-reader.cc

```diff
@@ -886,6 +886,36 @@ TEST_CASE_METHOD(
       "DenseReader: Memory budget is too small to open array");
 }
 
+TEST_CASE_METHOD(
+    CDenseFx,
+    "Dense reader: budget too small for tile offsets",
+    "[dense-reader][budget-too-small-tile-offsets]") {
+  // Create default array.
+  reset_config();
+  create_default_array_1d();
+
+  // Write a fragment.
+  int subarray[] = {1, NUM_CELLS};
+  std::vector<int> data(NUM_CELLS);
+  std::iota(data.begin(), data.end(), 1);
+  uint64_t data_size = data.size() * sizeof(int);
+  write_1d_fragment(subarray, data.data(), &data_size);
+
+  total_budget_ = "600";
+  tile_upper_memory_limit_ = "200";
+  update_config();
+
+  // Try to read.
+  int data_r[NUM_CELLS] = {0};
+  uint64_t data_r_size = sizeof(data_r);
```

Comment:
Can you add some more comments to make the test clearer? I don't understand the calculations here.

Also can we add a less trivial test too where we write more than 1 fragment?

Reply:
What computation? This just computes the size of the user buffer.

Reply:
I meant the calculations in the test. It's wasn't evident to untrained eyes why a budget of 600 previously was not enough. Thank you for adding those comments.

One last question, does `tile_upper_memory_limit_` play a role here? Could it be completely omitted so that we test what's actually the case, what happens when we only set `total_budget` and leave that one as default?

Reply:
It could be removed from there.